lib/krikri/software_agent.rb

```diff
@@ -30,6 +30,38 @@ def log(priority, msg)
     end
 
     ##
+    # Store this agent's generator activity, which is the activity that
+    # produced the target entities upon which the current agent will operate.
+    #
+    # It is assumed that the class that includes SoftwareAgent will define
+    # class methods .entity_behavior and .generator_entity_behavior, which
+    # return the class of the appropriate behavior.
+    #
+    # We look for either `generator_activity' or `generator_uri' in the `opts'
+    # keys.  `generator_activity' must be a Krikri::Activity object, and
+    # `generator_uri' can be a string or RDF::URI
+    #
+    # @see Krikri::Mapper::Agent
+    # @see Krikri::Harvester
+    #
+    def set_generator_activity!(opts)
```

Comment:
This is here because we used to only be passed a generator URI when instantiating an agent, but now the agent needs a property representing an activity; because that has a behavior that we need, which tells us the entities that were generated.

`initialize` was always passed `generator_uri`, for mappers, at least, but we need an activity object.

If an agent like a harvester doesn't have a generator URI, this does nothing.

Sure, it could be trimmed down by removing the handling of `generator_activity`, but I probably put that there because I could imagine a good use for it.